### PR TITLE
Cleanup var references

### LINKED ISSUE #96: Adapt to terraform v0.12 language changes
**How to categorize this issue?**

/area open-source
/kind cleanup
/priority normal
/platform openstack

**What would you like to be added**:
provider-openstack needs an adaptation of the terraform configuration to v0.12. For provider-aws this is done with this PR - https://github.com/gardener/gardener-extension-provider-aws/pull/111. 

**Why is this needed**:
Currently the terraformer run is only omitting warnings but in a future version of terraform, the warnings will be switched to errors.

### PR DESCRIPTION
**Pull request by afritzler**
**How to categorize this PR?**
/kind cleanup
/priority normal
/platform openstack

**What this PR does / why we need it**:
Remove warnings for Terraform v0.12.x runs.

**Which issue(s) this PR fixes**:
Fixes #96 

**Release note**:
```improvement operator
Until now `provider-openstack` was maintaining a Terraform configuration which is both `v0.12` and `v0.11` compatible. The Terraform configuration is now adapted to the new Terraform language which makes it Terraform `v0.11` incompatible.
```

### CODE REVIEW

File: charts/internal/openstack-infra/templates/main.tf [hunk 1]
```diff
@@ -43,7 +43,7 @@ resource "openstack_networking_subnet_v2" "cluster" {
 
 resource "openstack_networking_router_interface_v2" "router_nodes" {
   router_id = "{{ required "router.id is required" $.Values.router.id }}"
```
Comment (ialidzhikov) [inline charts/internal/openstack-infra/templates/main.tf:45]: I think this PR is not fixing all of the usage of the old syntax. For example 
```
  router_id = "{{ required "router.id is required" $.Values.router.id }}"
```

can be 
```
  router_id = "${openstack_networking_router_v2.router.id}"
```

which is still deprecated usage.

Comment (afritzler) [reply charts/internal/openstack-infra/templates/main.tf:45]: Good point! The `DefaultRouterID` should be fixed now by 
```
DefaultRouterID = "openstack_networking_router_v2.router.id"
```

Comment (ialidzhikov) [reply charts/internal/openstack-infra/templates/main.tf:45]: Hmm, still:
```
  router_id = "openstack_networking_router_v2.router.id"
```

!=


```
  router_id = openstack_networking_router_v2.router.id
```

We need the quotes only when and existing ID is specified. We don't need the quotes when we use the tf expresison.

Comment (afritzler) [reply charts/internal/openstack-infra/templates/main.tf:45]: Actually we don't need the quotes in the rendered helm output at all.

Comment (ialidzhikov) [reply charts/internal/openstack-infra/templates/main.tf:45]: When routerID is specified, you can pass the chart value explicitly quoted. At least this was my approach in the respective PR for provider-aws.

**Comment by rfranzke** [comment]
@afritzler can you please update the PR?

**Comment by afritzler** [comment]
Done.

File: charts/internal/openstack-infra/templates/main.tf (hunk 1 above)
Comment (afritzler) [reply charts/internal/openstack-infra/templates/main.tf:56]: Not sure if I am missing something but this is the same approach as in your aws provider PR https://github.com/ialidzhikov/gardener-extension-provider-aws/blob/25b75a6ae4c08a850c04d68c6a8516fbbb6deb12/pkg/controller/infrastructure/actuator_reconcile.go#L111

File: charts/internal/openstack-infra/templates/main.tf [hunk 2]
```diff
@@ -3,8 +3,8 @@ provider "openstack" {
   domain_name = "{{ required "openstack.domainName is required" .Values.openstack.domainName }}"
   tenant_name = "{{ required "openstack.tenantName is required" .Values.openstack.tenantName }}"
   region      = "{{ required "openstack.region is required" .Values.openstack.region }}"
-  user_name   = "${var.USER_NAME}"
-  password    = "${var.PASSWORD}"
+  user_name   = var.USER_NAME
+  password    = var.PASSWORD
```
Comment (ialidzhikov) [inline charts/internal/openstack-infra/templates/main.tf:7]: Can you adapt also `variables.tf`?

File: charts/internal/openstack-infra/templates/main.tf (hunk 1 above)
Comment (ialidzhikov) [reply charts/internal/openstack-infra/templates/main.tf:56]:
```
  router_id = "openstack_networking_router_v2.router.id"
```

should fail that there is no existing router with this id.

**Comment by ialidzhikov** [comment]
@afritzler , assuming that you are also busy with other things, I opened #112 where I applied also the changes requested in this PR and added you as co-author. I hope that this is fine for your.

**Pull request closed without merging**